datafusion/sqllogictest/test_files/push_down_filter_join.slt

```diff
@@ -0,0 +1,612 @@
+----
+11 a 11 z
+
+query ITIT
+SELECT * FROM t1 INNER JOIN t2 ON t1.t1_id = t2.t2_id AND t1.t1_name IS NULL ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+99 NULL 99 u
+
+query ITIT
+SELECT * FROM t1 INNER JOIN t2 ON t1.t1_id = t2.t2_id AND t2.t2_name IS NULL ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+22 b 22 NULL
+
+query ITIT
+SELECT * FROM t1 INNER JOIN t2 ON t1.t1_id = t2.t2_id AND t1.t1_name IS NULL ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+99 NULL 99 u
+
+query ITIT
+SELECT * FROM t1 INNER JOIN t2 ON t1.t1_id = t2.t2_id AND t2.t2_name IS NULL ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+22 b 22 NULL
+
+query ITIT
+SELECT * FROM t1 INNER JOIN t2 ON t1.t1_id = t2.t2_id AND t1.t1_id <= 22 ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+11 a 11 z
+22 b 22 NULL
+
+query ITIT
+SELECT * FROM t1 INNER JOIN t2 ON t1.t1_id = t2.t2_id AND t2.t2_id <= 22 ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+11 a 11 z
+22 b 22 NULL
+
+# Left Outer Join
+
+## `WHERE` clause
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+11 a 11 z
+22 b 22 NULL
+33 c NULL NULL
+44 d 44 x
+77 e NULL NULL
+88 NULL NULL NULL
+99 NULL 99 u
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id WHERE t1.t1_name = 'a' ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+11 a 11 z
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id WHERE t2.t2_name = 'z' ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+11 a 11 z
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id WHERE t1.t1_name IS NULL ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+88 NULL NULL NULL
+99 NULL 99 u
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id WHERE t2.t2_name IS NULL ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+22 b 22 NULL
+33 c NULL NULL
+77 e NULL NULL
+88 NULL NULL NULL
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id WHERE t1.t1_id <= 22 ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+11 a 11 z
+22 b 22 NULL
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id WHERE t2.t2_id <= 22 ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+11 a 11 z
+22 b 22 NULL
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id WHERE t1.t1_id IS NULL ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id WHERE t2.t2_id IS NULL ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+33 c NULL NULL
+77 e NULL NULL
+88 NULL NULL NULL
+
+## `ON` clause
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id AND t1.t1_name = 'a' ORDER BY t1_id, t1_name, t2_id, t2_name
+----
+11 a 11 z
+22 b NULL NULL
+33 c NULL NULL
+44 d NULL NULL
+77 e NULL NULL
+88 NULL NULL NULL
+99 NULL NULL NULL
+
+query ITIT
+SELECT * FROM t1 LEFT JOIN t2 ON t1.t1_id = t2.t2_id AND t2.t2_name = 'z' ORDER BY t1_id, t1_name, t2_id, t2_name
```

Comment:
I think the predicates here are "to simple" and a lot of the outer joins here will be converted to `Inner` joins by eliminate_outer_join rule and therefore these tests do not provide the desired coverage.